### PR TITLE
Added optional/default parameters for StreamWriter/StreamReader

### PR DESCRIPTION
**Pull request by jimdemis**
This resolves the CoreCLR side of [dotnet/corefx#8173](https://github.com/dotnet/corefx/issues/8173)

### CODE REVIEW

**Comment by jimdemis** [comment]
The CoreFX side: [dotnet/corefx#36959](https://github.com/dotnet/corefx/pull/36959)

**Comment by jkotas** [comment]
Could you please disable the outdated CoreFX test in https://github.com/dotnet/coreclr/blob/master/tests/CoreFX/CoreFX.issues.json to make the PR green

File: src/System.Private.CoreLib/shared/System/IO/StreamWriter.cs [hunk 1]
```diff
@@ -89,18 +89,26 @@ public StreamWriter(Stream stream, Encoding encoding, int bufferSize)
...
         {
-            if (stream == null || encoding == null)
+            if (stream == null)
             {
-                throw new ArgumentNullException(stream == null ? nameof(stream) : nameof(encoding));
+                throw new ArgumentNullException(nameof(stream));
+            }
+            if (encoding == null)
+            {
+                encoding = Encoding.UTF8;
```
Comment (justinvp) [inline src/System.Private.CoreLib/shared/System/IO/StreamWriter.cs]: Should this be `UTF8NoBOM`?

Comment (jimdemis) [reply src/System.Private.CoreLib/shared/System/IO/StreamWriter.cs]: /cc @JeremyKuhne

Comment (JeremyKuhne) [reply src/System.Private.CoreLib/shared/System/IO/StreamWriter.cs]: Yes, it should match what the simpler constructor would do- so `UTF8NoBOM`.

Comment (JeremyKuhne) [reply src/System.Private.CoreLib/shared/System/IO/StreamWriter.cs]: Outside of this it looks good to me.

**Comment by jimdemis** [comment]
According to failing builds, it looks like System.Text.Encoding is not nullable anymore:
```
error CS8625: Cannot convert null literal to non-nullable reference type
```
Is it intended? Should I change the constructor to nullable "Encoding?"
/cc @JeremyKuhne @safern

**Comment by stephentoub** [comment]
Thanks.  @JeremyKuhne, can you please review as well?

**Pull request merged**